### PR TITLE
feat: Lens Frames initial specification

### PR DESCRIPTION
**Pull request by defispartan**
Lens Frames are an implementation of the Open Frames standard to define structure of authenticated frame applications for Lens Protocol. Once finalized the spec will be proposed to the Lens developer community as an [LIP](https://github.com/lens-protocol/LIPS).

Lens modules provide unparalleled customization, interoperability, and attribution for onchain actions. This standard will enable modules (and any other Frames use case) to experience the same interoperability at the application level.

Frame servers can authenticate requests from Lens profiles 🤝 Lens apps can expose custom modules

- [x] Lens API authentication methods
- [x] Lens Client and React SDK authentication methods
- [x] Lens Frames specification README

### CODE REVIEW

File: demo-client-application/app/profile/page.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export default function ProfileComponent() {
+  const { data } = useSession();
+  if (!data) return null
+  return (
+    <div className="px-10 py-14 flex flex-col items-center">
+      {
+        data.type === "WITH_PROFILE" && data.profile && (
+          <Profile
+            handle={`${data.profile.handle?.namespace}/${data.profile.handle?.localName}`}
```
Comment (krzysu) [inline demo-client-application/app/profile/page.tsx]:
```suggestion
            handle={data.profile.handle?.fullHandle}
```

File: demo-client-application/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "next-themes": "^0.2.1",
+    "postcss": "8.4.29",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "tailwind-merge": "^1.14.0",
+    "tailwindcss": "3.3.3",
+    "tailwindcss-animate": "^1.0.7",
+    "typescript": "5.2.2",
+    "viem": "^1.15.0",
+    "wagmi": "^1.4.12"
```
Comment (krzysu) [inline demo-client-application/package.json]: with latest version of the lens react sdk you can use latest wagmi and view (v2)

File: demo-client-application/app/profile/page.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+} from '@lens-protocol/react-web'
+import { Profile } from '@lens-protocol/widgets-react'
+
+export default function ProfileComponent() {
+  const { data } = useSession();
+  if (!data) return null
+  return (
+    <div className="px-10 py-14 flex flex-col items-center">
+      {
+        data.type === "WITH_PROFILE" && data.profile && (
```
Comment (krzysu) [inline demo-client-application/app/profile/page.tsx]:
```suggestion
        data.type === SessionType.WithProfile && data.profile && (
```

use provided enums, SessionType you can import from "@lens-protocol/react-web"

File: demo-client-application/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@lens-protocol/metadata": "^1.0.5",
```
Comment (krzysu) [inline demo-client-application/package.json]: this package is not used at all

**Review by krzysu — COMMENTED** [review]
hi, I left just a few small improvements in the usage of the lens react sdk

**Comment by defispartan** [comment]
@krzysu Updated the demo client to use code from examples in Lens SDK, which resolved all of the comments

**Pull request merged**